WooCommercePOS/WooCommercePOS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -0,0 +1,18 @@
+import SwiftUI
+
+final class PointOfSaleDashboardViewModel: ObservableObject {
+    @Published var products: [Product]
+    @Published var productsInOrder: [Product] = []
+
+    init(products: [Product]) {
+        self.products = products
```

Comment:
nit: I understand that this is just part of the initial work that we can update later so feel free to keep it as it is - based on the initial design that allows search and without knowing whether products are loaded before entering POS mode, we probably want to fetch the products (with or without an API sync) on the POS view presentation within the POS scope (like this or a a different view model). This means that this view model is not taking in a list of products in the initializer, and the mock `products` can be initialized within the initializer itself.

```suggestion
    init() {
        self.products = ProductFactory.makeFakeProducts()
```

Reply:
I agree! I did approach it like you mention initially, but ultimately I decided to pass them through the initializer to reflect that these come from the first thing we find in the POS scope, which in this case would be the `PointOfSaleEntryPointView`. I'm not sure if we even need an "entry point view", or we could call the dashboard directly so I'll leave it as it is for now while we work on navigation and data layer 🙇